dark_rabbit/services/sender_worker.py

```diff
@@ -0,0 +1,36 @@
+import logging
+
+from odoo.addons.generic_background_service import AbstractBackgroundServiceWorker
+
+_logger = logging.getLogger(__name__)
+
+# POLLING_CYCLE_INTERVAL represents the timeout between event polling cycles
+POLLING_CYCLE_INTERVAL = 10  # seconds
+
+
+class DarkRabbitSenderWorker(AbstractBackgroundServiceWorker):
+    """This class represents service worker for single database.
+
+    It is responsible for reading info about connections,
+    and spawning rabbit sender in separate threads for each
+    'dark.rabbit.connection'.
+    """
+
+    def get_sleep_timeout(self):
+        # The time between polling cycles (Calls to run_service)
+        return POLLING_CYCLE_INTERVAL
+
+    def run_service(self):
+        with self.with_env() as env:
+
+            connections = env["dark.rabbit.connection"].search([])
+
+            for connection in connections:
+                events_for_send = env["dark.rabbit.outgoing.event"].search(
+                    [
+                        ("sent_at", "=", False),
+                        ("connection_id", "=", connection.id),
+                    ]
+                )
```

Comment:
Add limit for events to send. Let's send at most 100 events at single iteration.

Also, it is better to add following logic, that checks if some events were sent, then we check for more events immediately, otherwise we can sleep for Polling Cycle interval. The idea is, if we receive large amount of events, then we need to send them as fast as possible. When all events are sent, then worker can sleep for some time.

Additionally in these cycles, we have to check for the signal for shutdown of worker, which could be sent by service (for example if we shut down the server before update.